packages/vite/src/node/optimizer/scan.ts

```diff
@@ -218,8 +222,14 @@ function esbuildScanPlugin(
 
       // html types: extract script contents -----------------------------------
       build.onResolve({ filter: htmlTypesRE }, async ({ path, importer }) => {
+        const resolved = await resolve(path, importer)
+        if (!resolved) return
+        // It is possible for the scanner to scan html types in node_modules.
+        // If we can optimize this html type, skip it so it's handled by the
+        // bare import resolve, and recorded as optimization dep.
+        if (resolved.includes('node_modules') && canOptimize(resolved)) return
```

Comment:
I realized when we import something like `my-compoenent-lib/Test.vue` directly, the file is scanned as well, so deps imported within the Vue file is actually being prebundled, and surprisingly without errors.

Not sure if this is intentional, but for the case where we can optimize the file, we don't want this behaviour.